### DISCUSSION TITLE
How to recreate/find MP ids in your hdf5 train/val splits

### DISCUSSION THREAD

**Discussion by DeNeutoy** [ACEsuit/mace#502] [Q&A]
Hello!  In #297 the information you provided around the validation energy MAEs was very helpful. I was wondering if you could help clarify one thing:

I was wondering if it is possible to recreate the HDF5 data you have provided as your train/val split. The reason I want to do this is to retrieve the mp-ids for each batch - when I looked into this, it seems difficult because:

- In the HDF5 files, there is no reference to which .extxyz file it came from
- In the [dataset provided as training data in the MACE MP repository](https://github.com/ACEsuit/mace-mp/tree/main?tab=readme-ov-file#training-data),  there is no train/val split, so I assume you have used the [functionality in your pre-processing script here](https://github.com/ACEsuit/mace/blob/main/mace/tools/scripts_utils.py#L80) to create a random split.  

The reason I would like to recreate your dataset exactly is because we have computed some dispersion corrections on the original MPTraj dataset, but I would also like to use your exact data split to make the comparison between the models fairer. Do you see a way for this to be possible?

If not, do you think that this will matter, or have you observed low variance when using different randomly held out splits of MPTraj? I couldn't quite tell from your pre-processing code if you are holding out entire trajectories from MPtraj for your validation set also, or if you consider each trajectory point to be independent from a dataset split perspective.

**Comment by ilyes319** [comment]
Hey,

To recreate the files, you need to use the pre-processing (see doc [here](https://mace-docs.readthedocs.io/en/latest/guide/multipreprocessing.html)). All the point of the trajectories are independent. I think to get the id, you should redo the split by hand, and then process each files (train and valid) indepdently. I don't think the training is sensitive to the split.
